File: varuna-backend/app/api/v1/weather.py
"""
Weather API Endpoints
"""

from typing import List
from fastapi import APIRouter, Query
from app.schemas.map import WeatherIntelligence, WeatherForecastItem
from app.services.weather_service import weather_service

router = APIRouter()


@router.get("/intelligence", response_model=WeatherIntelligence)
async def get_weather_intelligence(
    lat: float = Query(17.38, description="Latitude"),
    lng: float = Query(83.25, description="Longitude"),
):
    return await weather_service.get_weather(lat, lng)


@router.get("/forecast", response_model=List[WeatherForecastItem])
async def get_weather_forecast(
    lat: float = Query(17.38, description="Latitude"),
    lng: float = Query(83.25, description="Longitude"),
):
    weather = await weather_service.get_weather(lat, lng)
    return weather.forecast

File: Varuna-Frontend/src/data/api/weather.ts
/**
 * VARUNA Weather API
 */

import { apiClient } from './client';
import { WeatherIntelligence, WeatherForecastItem } from '../../domain/models/mapIntelligence';

export const weatherApi = {
  async getWeatherIntelligence(lat: number, lng: number): Promise<WeatherIntelligence> {
    return apiClient.get<WeatherIntelligence>('/weather/intelligence', {
      params: { lat, lng },
    });
  },

  async getForecast(lat: number, lng: number, days: number = 3): Promise<WeatherForecastItem[]> {
    return apiClient.get<WeatherForecastItem[]>('/weather/forecast', {
      params: { lat, lng, days },
    });
  },
};
